api/hypershift/v1beta1/hostedcluster_types.go

```diff
@@ -1248,9 +1248,72 @@ const (
 // +kubebuilder:validation:Enum=Karpenter
 type Provisioner string
 
+// Configures when and how to scale down cluster nodes.
+type ScaleDownConfig struct {
+	// enabled determines whether the Cluster Autoscaler should scaled down this cluster.
+	Enabled bool `json:"enabled"`
+
+	// delayAfterAdd sets how long after scale up the scale down evaluation resumes
+	// +kubebuilder:validation:Pattern=([0-9]*(\.[0-9]*)?[a-z]+)+
+	// +kubebuilder:validation:MaxLength=100
+	// +optional
+	DelayAfterAdd *string `json:"delayAfterAdd,omitempty"`
+
+	// delayAfterDelete sets how long after node deletion, scale down evaluation resumes, defaults to scan-interval
+	// +kubebuilder:validation:Pattern=([0-9]*(\.[0-9]*)?[a-z]+)+
+	// +kubebuilder:validation:MaxLength=100
+	// +optional
+	DelayAfterDelete *string `json:"delayAfterDelete,omitempty"`
+
+	// delayAfterFailure sets how long after a scale down failure, scale down evaluation resumes
+	// +kubebuilder:validation:Pattern=([0-9]*(\.[0-9]*)?[a-z]+)+
+	// +kubebuilder:validation:MaxLength=100
+	// +optional
+	DelayAfterFailure *string `json:"delayAfterFailure,omitempty"`
+
+	// unneededTime establishes how long a node should be unneeded before it is eligible for scale down
+	// +kubebuilder:validation:Pattern=([0-9]*(\.[0-9]*)?[a-z]+)+
+	// +kubebuilder:validation:MaxLength=100
+	// +optional
+	UnneededTime *string `json:"unneededTime,omitempty"`
+
+	// utilizationThreshold determines the node utilization level, defined as sum of requested resources divided by capacity, below which a node can be considered for scale down
+	// +kubebuilder:validation:Pattern=(0.[0-9]+)
+	// +kubebuilder:validation:MaxLength=100
+	// +optional
+	UtilizationThreshold *string `json:"utilizationThreshold,omitempty"`
+}
+
+// ExpanderString contains the name of an expander to be used by the cluster autoscaler.
+// +kubebuilder:validation:Enum=LeastWaste;Priority;Random
+type ExpanderString string
+
+// These constants define the valid values for an ExpanderString
+const (
+	LeastWasteExpander ExpanderString = "LeastWaste" // Selects the node group with the least idle resources.
+	PriorityExpander   ExpanderString = "Priority"   // Selects the node group with the highest priority.
+	RandomExpander     ExpanderString = "Random"     // Selects a node group randomly.
+)
+
 // ClusterAutoscaling specifies auto-scaling behavior that applies to all
 // NodePools associated with a control plane.
 type ClusterAutoscaling struct {
+	// scaleDown configures the behavior of the Cluster Autoscaler scale down operation.
+	//
+	// +optional
+	ScaleDown *ScaleDownConfig `json:"scaleDown,omitempty"`
+
+	// balancingIgnoredLabels sets "--balancing-ignore-label <label name>" flag on cluster-autoscaler for each listed label.
```

Comment:
This field only makes sense when using the balance similar nodes option right? So, can we validate that this list is empty when that feature is disabled?

Reply:
If we hardcode it, then that's fine, lets not expose it